flow/project.py

```diff
@@ -418,11 +435,21 @@ def __init__(self, cmd, pre=None, post=None, directives=None):
     def __str__(self):
         return "{type}(cmd='{cmd}')".format(type=type(self).__name__, cmd=self._cmd)
 
-    def eligible(self, job):
-        "Eligible, when all pre-conditions are true and at least one post-condition is false."
-        pre = all(cond(job) for cond in self._prereqs)
+    def eligible(self, job, ignore_conditions='none'):
+        """Eligible, when all pre-conditions are true and at least one post-condition is false,
+        or corresponding conditions are ignored
+        """
+        if ignore_conditions in _IGNORE_CONDITIONS.keys():
+            ignore_conditions = _IGNORE_CONDITIONS[ignore_conditions]
+        else:
+            raise ValueError(
+                "The ignore_conditions argument of FlowProject.run() must be a string in "
+                "'none', 'pre', 'post' or 'all'")
+        pre = (not len(self._prereqs)) or (ignore_conditions & IgnoreConditions.PRE) \
```

Comment:
Is the `len` check here as an optimization? If so, I'd include a comment to that effect since you can remove that check and the conditions are still equivalent (because `all([])` evaluates to `True`).